domdiv/main.py

```diff
@@ -1359,6 +1430,54 @@ def filter_sort_cards(cards, options):
                          card_tag=set_tag)
                 cards.append(c)
 
+    # Take care of any --only-type-xxx requirements
+    if options.only_type_any or options.only_type_all:
+        # First make a dictionary for easier lookup of Type name used by the program
+        # The index in each case is lower case for easier matching
+        # The value in each case is the type index as used in types_en_us.json
+        types_lookup = defaultdict(dict)
+        for x in Card.type_names:
+            types_lookup[x.lower()] = x
+            types_lookup[Card.type_names[x].lower()] = x
+
+        # Start the valid lists
+        type_unknown = []
+        type_known_any = []
+        type_known_all = []
+
+        # Assemble a list of valid "any" types.  Options are already lowercase.
+        for x in options.only_type_any:
+            if x in types_lookup:
+                type_known_any.append(types_lookup[x])
+            else:
+                type_unknown.append(x)
+
+        # Assemble a list of valid "all" types. Options are already lowercase.
+        for x in options.only_type_all:
+            if x in types_lookup:
+                type_known_all.append(types_lookup[x])
+            else:
+                type_unknown.append(x)
+
+        # Indicate if unknown types are given
+        if type_unknown:
+            print(("Error - unknown type(s): {}".format(", ".join(type_unknown))))
```

Comment:
As usual, I think I'd rather have this quit with an error here.

Reply:
Done.  And I fixed a few others too in the arguments area.